src/main/java/org/wso2/carbon/connector/ai/AIConverser.java

```diff
@@ -0,0 +1,39 @@
+package org.wso2.carbon.connector.ai;
+
+import org.apache.axis2.AxisFault;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.SynapseException;
+import org.apache.synapse.commons.json.JsonUtil;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.wso2.carbon.connector.ai.connection.AIConnection;
+import org.wso2.carbon.connector.ai.constants.AIConstants;
+import org.wso2.carbon.connector.ai.model.prompt.AIConverserAgentModel;
+import org.wso2.carbon.connector.ai.util.AIUtils;
+import org.wso2.carbon.connector.core.AbstractConnector;
+import org.wso2.carbon.connector.core.ConnectException;
+import org.wso2.carbon.connector.core.connection.ConnectionHandler;
+
+/**
+ * This class handles execution of a given prompt with AI.
+ */
+public class AIConverser extends AbstractConnector {
+
+    @Override
+    public void connect(MessageContext messageContext) throws ConnectException {
+        try {
+            org.apache.axis2.context.MessageContext msgContext = ((Axis2MessageContext) messageContext).getAxis2MessageContext();
+
+            ConnectionHandler handler = ConnectionHandler.getConnectionHandler();
+            AIConnection aiConnection = (AIConnection) handler
+                    .getConnection(AIConstants.CONNECTOR_NAME, AIUtils.getConnectionName(messageContext));
+            AIConverserAgentModel agent = AIUtils.getAIConverserAgent(messageContext);
+            agent.processRequest(aiConnection);
+            JsonUtil.getNewJsonPayload(msgContext, agent.getResponse(), true, true);
+            msgContext.setProperty(AIConstants.MESSAGE_TYPE_STRING, AIConstants.JSON_CONTENT_TYPE);
+            msgContext.setProperty(AIConstants.CONTENT_TYPE_STRING, AIConstants.JSON_CONTENT_TYPE);
+        } catch (AxisFault e) {
+            handleException("JsonUtil error.", e, messageContext);
```

Comment:
Can we add a more descriptive error message here?

Reply:
Updated error messege to : "Error processing JSON payload: Invalid response format."